From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Daniel Kiper <daniel.kiper@oracle.com>
Date: Thu, 6 Dec 2018 13:38:15 +0100
Subject: [PATCH] verifiers: Xen fallout cleanup

Xen fallout cleanup after commit ca0a4f689 (verifiers: File type for
fine-grained signature-verification controlling).

Signed-off-by: Daniel Kiper <daniel.kiper@oracle.com>
Reviewed-by: Ross Philipson <ross.philipson@oracle.com>
---
 grub-core/loader/i386/xen.c | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/grub-core/loader/i386/xen.c b/grub-core/loader/i386/xen.c
index 07a4837c5..071b530d7 100644
--- a/grub-core/loader/i386/xen.c
+++ b/grub-core/loader/i386/xen.c
@@ -647,10 +647,10 @@ grub_cmd_xen (grub_command_t cmd __attribute__ ((unused)),
 
   grub_xen_reset ();
 
-  grub_create_loader_cmdline (argc - 1, argv + 1,
-			      (char *) xen_state.next_start.cmd_line,
-			      sizeof (xen_state.next_start.cmd_line) - 1);
-  err = grub_verify_string (xen_state.next_start.cmd_line, GRUB_VERIFY_MODULE_CMDLINE);
+  err = grub_create_loader_cmdline (argc - 1, argv + 1,
+				    (char *) xen_state.next_start.cmd_line,
+				    sizeof (xen_state.next_start.cmd_line) - 1,
+				    GRUB_VERIFY_KERNEL_CMDLINE);
   if (err)
     return err;
 
@@ -918,9 +918,9 @@ grub_cmd_module (grub_command_t cmd __attribute__ ((unused)),
   if (err)
     goto fail;
 
-  grub_create_loader_cmdline (argc - 1, argv + 1,
-			      get_virtual_current_address (ch), cmdline_len);
-  err = grub_verify_string (get_virtual_current_address (ch), GRUB_VERIFY_MODULE_CMDLINE);
+  err = grub_create_loader_cmdline (argc - 1, argv + 1,
+				    get_virtual_current_address (ch), cmdline_len,
+				    GRUB_VERIFY_MODULE_CMDLINE);
   if (err)
     goto fail;
 
